Type improvements to landing.py and panel.py

VasigaranAndAngel opened the pull request:
### Summary
- few type improvements to both landing.py and panel.py
- renamed a method in the `PanelModal` class (in panel.py) and updated all references in other files
<!--
^^^ Summarize the changes done and why they were done above.

By submitting this pull request, you certify that you have read the
[CONTRIBUTING.md](https://github.com/TagStudioDev/TagStudio/blob/main/CONTRIBUTING.md).

IMPORTANT FOR FEATURES: Please verify that a feature request or some other form
of communication with maintainers was already conducted in terms of approving.

Thank you for your eagerness to contribute!
-->

### Tasks Completed

<!-- No requirements, just context for reviewers. -->

-   Platforms Tested:
    -   [x] Windows x86
    -   [ ] Windows ARM
    -   [ ] macOS x86
    -   [ ] macOS ARM
    -   [ ] Linux x86
    -   [ ] Linux ARM
    <!-- If an unspecified platform was tested, please add it here -->
-   Tested For:
    -   [x] Basic functionality
    -   [ ] PyInstaller executable <!-- Not necessarily required, but appreciated! -->
    <!-- If other important criteria was tested for, please add it here -->

Computerdores commented on src/tagstudio/qt/modals/build_tag.py:
```diff
@@ -169,7 +169,7 @@ def __init__(self, library: Library, tag: Tag | None = None) -> None:
         tsp.tag_chosen.connect(lambda x: self.add_parent_tag_callback(x))
         self.add_tag_modal = PanelModal(tsp)
-        self.add_tag_modal.setTitle(Translations["tag.parent_tags.add"])
+        self.add_tag_modal.set_title(Translations["tag.parent_tags.add"])
         self.add_tag_modal.setWindowTitle(Translations["tag.parent_tags.add"])
```
These should just be passed to the PanelModal constructor

Computerdores reviewed (commented):
Note: I the comments I added aren't issues with what you did, but issues with the code you touched that might as well be fixed now while the code is being changed anyways

Computerdores reviewed (commented):
I am sorry if this was unclear, but what I intended to say was that both the title and the window title could be moved to the constructor.
Unless it was clear enough and you didn't do it for another reason?

VasigaranAndAngel commented:
> I am sorry if this was unclear, but what I intended to say was that both the title and the window title could be moved to the constructor. Unless it was clear enough and you didn't do it for another reason?

I changed the code to pass the title to the constructor as you suggested. since the `set_title` method was no longer used elsewhere and seemed unnecessary, i removed it. i don't see a need to change the title after creating a panel window, and if it's requred, the `title_widget` can still be accessed directly. that's why i removed it.

CyanVoxel commented:
Thank you for the changes, and @Computerdores for the reviews!
